### PR TITLE
arm/tlsr82: pwm driver optimize and add pulse count support.

### PR DESCRIPTION
**Pull request by CV-Bowen**
## Summary
1. add pulse count support for pwm0;
2. add more detailed config for pwm;
3. pwm configuration and start process optimize;
4. tlsr82/Kconfig format;

## Impact
arm tlsr82xx chip pwm

## Testing
local test with pwm example
1. pwm max/min duty test;
2. pwm max/min frequency test;
3. pwm0 pulse count test;

### CODE REVIEW

**Review by xiaoxiang781216 — APPROVED** [review]

File: arch/arm/src/tlsr82/Kconfig [hunk 1]
```diff
@@ -250,6 +250,53 @@ menuconfig TLSR82_USB
 menuconfig TLSR82_PWM
 	bool "PWM Configuration"
 	default n
+	select PWM
+
+if TLSR82_PWM
+
+# PWM0 configuration
+
+	config TLSR82_PWM0
```
Comment (pkarashchenko) [inline arch/arm/src/tlsr82/Kconfig:259]: I think other paces do not have such alignment. Maybe we can remove TAB suffix in this case as well?

Comment (CV-Bowen) [reply arch/arm/src/tlsr82/Kconfig]: Ack and Done.

File: arch/arm/src/tlsr82/tlsr82_pwm.c [hunk 2]
```diff
@@ -589,16 +507,13 @@ static int pwm_cfg_check(struct tlsr82_pwmtimer_s *priv)
...
- *   APB1 or 2 clocking for the GPIOs has already been configured by the RCC
- *   logic at power up.
- *
  ****************************************************************************/
 
 static int pwm_setup(struct pwm_lowerhalf_s *dev)
 {
   struct tlsr82_pwmtimer_s *priv = (struct tlsr82_pwmtimer_s *)dev;
   uint32_t pincfg;
+  int ret = OK;
```
Comment (pkarashchenko) [inline arch/arm/src/tlsr82/tlsr82_pwm.c:516]: seems like init value is not needed. Let's remove `= OK;` or change `return OK;` to `return ret;`

**Review by pkarashchenko — APPROVED** [review]

File: arch/arm/src/tlsr82/tlsr82_pwm.c (hunk 2 above)
Comment (CV-Bowen) [reply arch/arm/src/tlsr82/tlsr82_pwm.c:516]: Done

**Review by pkarashchenko — APPROVED** [review]

**Pull request merged**